pageserver/src/walingest.rs

```diff
@@ -277,8 +277,8 @@ impl WalIngest {
             .ingest_batch(interpreted.batch, &self.shard, ctx)
             .await?;
 
-        // If checkpoint data was updated, store the new version in the repository
-        if self.checkpoint_modified {
+        // If checkpoint data was updated, store the new version in the repository (on shard zero only)
+        if self.checkpoint_modified && self.shard.is_shard_zero() {
```

Comment:
I'm not sure about this.
`WalIngest::new` reads the checkpoint from storage and then uses it throughout. Looking at uses of `WalIngest::checkpoint`, I think we rely on having an up to date checkpoint.

---

Also, I don't see the need for this change. We could cut down on some records going to all pageservers if this is correct, but from my experiments said records don't add up to much and we are going to compress them anyway.